nemo_rl/models/generation/vllm.py

```diff
@@ -84,6 +85,30 @@ def __repr__(self) -> str:
         """
         return f"{self.__class__.__name__}"
 
+    def _get_vllm_guided_decoding_params(
+        self, guided_decoding_config: Optional[GuidedDecodingConfig]
+    ) -> Optional["GuidedDecodingParams"]:  # noqa: F821
+        """Get the guided decoding parameters for vLLM."""
+        from vllm.sampling_params import GuidedDecodingParams
+
+        if guided_decoding_config is None:
+            return None
+
+        if guided_decoding_config["mode"] == "json":
+            return GuidedDecodingParams(json=guided_decoding_config["json"])
+        elif guided_decoding_config["mode"] == "regex":
+            return GuidedDecodingParams(regex=guided_decoding_config["regex"])
+        elif guided_decoding_config["mode"] == "choice":
+            return GuidedDecodingParams(choice=guided_decoding_config["choice"])
+        elif guided_decoding_config["mode"] == "grammar":
+            return GuidedDecodingParams(grammar=guided_decoding_config["grammar"])
+        elif guided_decoding_config["mode"] == "json_object":
+            return GuidedDecodingParams(json_object=True)
+        else:
+            raise ValueError(
+                f"Unsupported guided decoding mode: {guided_decoding_config['mode']}"
+            )
```

Comment:
I would refactor this into a lookup to a `dict` or `match-case`. IIRC, ruff has a checking for excessively `if-elif-else` clause.